README.md

```diff
@@ -509,6 +509,42 @@ Will serialize/deserialize to this JSON:
 
 As with `SequenceField`, values will automatically be `trim()`ed unless `trim: false` is specified in the attribute's argument list.
 
+### Date and Time fields
+
+`DateTime` and `DateTimeImmutable` fields can also be serialized, and you can control how they are serialized using the `DateField` attribute.  It has two arguments, which may be used individually or together.  Specifying neither is the same as not specifying the `DateField` attribute at all.
+
+#### `format`
+
+This argument lets you specify the format that will be used when serializing.  It may be any string accepted by PHP's [date_format syntax](), including one of the various constants defined on `DateTimeInterface`.  If not specified, the default format is `RFC3339_EXTENDED`, or `Y-m-d\TH:i:s.vP`.  While not the most human-friendly, it is the default format used by Javascript/JSON so makes for reasonable compatibility.
+
+On deserializing, the `format` has no effect.  Serde will pass the string value to a `DateTime` or `DateTimeImmutable` constructor, so any format recognized by PHP will be parsed according to PHP's standard [date-parsing rules]().
```

Comment:
Just a note from production... this goes with `be conservative in what you do, be liberal in what you accept from others` but in reality in most of the projects I have seen there is a tendency to provide one format that is supported and strictly validated as the PHP can parse too much of weird inputs which are really undesired such as `2023-02-29`, `today`, `+-1 day` which are from the point of consuming API really undesired.

So I would either make it strict by default, or optionally, for example with additional `inputFormat` parameter.

Reply:
Well, what the code does now is just throw the string at `new DateTime()` and cross its fingers.  So tightening the list at all would require more work.  I'm also not sure off hand if you can tell DateTime to parse only a certain format, and I'm not really inclined to try and build my own date-format-validating logic.  That sounds really hard. 😄 

I agree Postel's Law is generally a bad idea, but in this case anything else requires a lot more work.

Reply:
Ah, there is https://www.php.net/manual/en/datetimeimmutable.createfromformat.php

Still not entirely sure if I want to go there.  That would very tightly restrict the possible input values; I'm not sure if the best way to allow an opt-in "be permissive" option then, which I would still want to support.

That may be more appropriate for a follow-up.

Reply:
I looked into it a bit more, and there's a problem.  `createFromFormat()` will fill in "missing" values in the output with the current time.  So if you have a Y-m-d string of `2022-07-04T00:00:00` and create a new DTI using the constructor, you get that timestamp.  If you use `createFromFormat()`, you get that date but the time is filled in with whenever you ran the code.

This... honestly feels like a PHP bug to me.  It's not something I'm going to deal with right now, though, so I am going to leave a note in the code and otherwise skip it for now.